tests/std/tests/P0645R10_text_formatting_custom_formatting/test.cpp

```diff
@@ -57,7 +57,7 @@ struct not_const_formattable_type {
 
 template <>
 struct std::formatter<basic_custom_formattable_type, char> {
-    basic_format_parse_context<char>::iterator parse(basic_format_parse_context<char>& parse_ctx) {
+    constexpr basic_format_parse_context<char>::iterator parse(basic_format_parse_context<char>& parse_ctx) {

```

Comment:
How is this change conforming if it forces user-provided formatters to have `constexpr` here? I don't see any such requirements in WG21-N4964 \[format.formatter.spec\], and this causes the example in /6 to fail to compile.

I do see the Standardese in \[format.fmt.string\]/3, I just don't understand how this is implementable.

Reply:
Some more clarification: this change doesn't add a general requirement that `parse` is `constexpr` - we did that when we first implemented `basic_format_string` - it instead removes a special case in which we don't require `parse` to be `constexpr` only when the format specifier in the format string is empty . This was a libfmt feature (not calling `parse` when the format specifier is empty) that was never standard, and indeed was rejected by LEWG when it was proposed for standardization. I'm not sure how the feature found its way into our implementation.